www/include/configuration/configObject/command/minHelpCommand.php

```diff
@@ -87,14 +94,18 @@
 $command = escapeshellcmd($command);
 
 $tab = explode(' ', $command);
-if (realpath($tab[0])) {
-    $command = realpath($tab[0]) . ' ' . $plugin . ' ' . $mode . ' --help';
-} else {
-    $command = $tab[0] . ' ' . $plugin . ' ' . $mode . ' --help';
-}
+$commandPath = realpath($tab[0]) === false ? $tab[0] : realpath($tab[0]);
+
+// Exec command only if located in allowed directories
+$dbResult = $pearDB->query('SELECT `resource_line` FROM `cfg_resource`');
+$allowedPaths = $dbResult->fetchAll(\PDO::FETCH_COLUMN);
 
-$stdout = shell_exec($command . " 2>&1");
-$msg = str_replace("\n", "<br />", $stdout);
+$msg = "Command not allowed";
+if (preg_match('#(' . implode('|', $allowedPaths) . ')#', $commandPath)) {
+    $command = $commandPath . ' ' . $plugin . ' ' . $mode . ' --help';
+    $stdout = shell_exec($command . " 2>&1");
```

Comment:
Please protect the command using **escapeshellcmd**

Reply:
It was already there but I moved it right before the exec